release-23.2: go.mod: bump Pebble to e406c9cdb433

RaduBerinde opened the pull request:
This is a no-op, we're just doing it so the dep sha is up to date (see #125713).

https://github.com/cockroachdb/pebble/commit/e406c9cd Revert "go: bump cockroachdb/errors to v1.11.3"
https://github.com/cockroachdb/pebble/commit/6803ae85 go: bump cockroachdb/errors to v1.11.3

Fixes: #125711
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/125886)
<!-- Reviewable:end -->

RaduBerinde committed:
release-23.2: go.mod: bump Pebble to e406c9cdb433

This is a no-op, we're just doing it so the dep sha is up to date (see #125713).

https://github.com/cockroachdb/pebble/commit/e406c9cd Revert "go: bump cockroachdb/errors to v1.11.3"
https://github.com/cockroachdb/pebble/commit/6803ae85 go: bump cockroachdb/errors to v1.11.3

Fixes: #125711
Release note: none
